be/src/formats/parquet/column_reader.cpp

```diff
@@ -246,18 +249,73 @@ class MapColumnReader : public ColumnReader {
     std::unique_ptr<ColumnReader> _value_reader;
 };
 
+class StructColumnReader : public ColumnReader {
+public:
+    explicit StructColumnReader(const ColumnReaderOptions& opts) {}
+    ~StructColumnReader() override = default;
+
+    Status init(const ParquetField* field, std::vector<std::unique_ptr<ColumnReader>> child_readers) {
+        _field = field;
+        _child_readers = std::move(child_readers);
+        return Status::OK();
+    }
+
+    Status prepare_batch(size_t* num_records, ColumnContentType content_type, vectorized::Column* dst) override {
+        vectorized::NullableColumn* nullable_column = nullptr;
+        vectorized::StructColumn* struct_column = nullptr;
+        if (_field->is_nullable) {
+            DCHECK(dst->is_nullable());
+            nullable_column = down_cast<vectorized::NullableColumn*>(dst);
+            DCHECK(nullable_column->mutable_data_column()->is_struct());
+            struct_column = down_cast<vectorized::StructColumn*>(nullable_column->mutable_data_column());
+        } else {
+            DCHECK(dst->is_struct());
+            struct_column = down_cast<vectorized::StructColumn*>(dst);
+        }
+
+        vectorized::Columns fields_column = struct_column->fields_column();
+
+        DCHECK_EQ(fields_column.size(), _child_readers.size());
+
+        for (size_t i = 0; i < fields_column.size(); i++) {
+            vectorized::Column* child_column = fields_column.at(i).get();
+            RETURN_IF_ERROR(_child_readers.at(i)->prepare_batch(num_records, content_type, child_column));
```

Comment:
I don't think so, I faced many times that access array out of bounds.

Reply:
The key point of accessing an array out of bounds is the logical problem, not how you access it.
No matter which way you use, you can't avoid out-of-bound, even with `at(i)`.
In my opinion, you should avoid the logical problem itself.
In addition, 1. you don't catch the exception. 2. most of the time, `i` must be valid because it is in the loop generated by the container size.